src/Modules/CalcPerform.lua

```diff
@@ -54,8 +54,13 @@ local function getTriggerActionTriggerRate(baseActionCooldown, env, breakdown, f
 	end
 
 	local modActionCooldown = cooldownOverride or (baseActionCooldown / icdr)
-	local rateCapAdjusted = m_ceil(modActionCooldown * data.misc.ServerTickRate) / data.misc.ServerTickRate
-	local extraICDRNeeded = m_ceil((modActionCooldown - rateCapAdjusted + data.misc.ServerTickTime) * icdr * 1000)
+	local rateCapAdjusted = modActionCooldown
+	local extraICDRNeeded = 0
+	if not storesMultipleUses then
```

Comment:
I realize now that this calculation is not entirely correct. It is possible to get rate cap adjusted even with multiple uses, if the cast rate is large enough to impact the _last_ use. 

Also, I don't know if adding this is controversial or not. I don't logically see why there would be a rate cap when there are multiple uses (since the next use can start ticking immediately), and the tests run in this reddit thread supports that logic:

https://www.reddit.com/r/pathofexile/comments/zse0mo/impending_doom_and_cdr_testing/

Reply:
I don't understand how vixen's has multiple charges I understand the logic behind multiple use skills not being tick adjusted though.

This likely affects seismic trap cooldowns as well then so tagging @Edvinas-Smita

Reply:
Vixen's Entrapment is bound by the rate cap (single use with 0.25 second cooldown), but Doom Blast itself is not (three uses with 0.15 second cooldown).

Vixen's cooldown calculation is called from CalcPerform.lua:3138
`local maxVixenTriggerRate = getTriggerActionTriggerRate(0.25, env)`

Doom Blast cooldown calculation is called from CalcPerform.lua:3147 (with last argument set to true)
`output.ActionTriggerRate = getTriggerActionTriggerRate(env.player.mainSkill.skillData.cooldown, env, breakdown, false, false, true)`

Reply:
I see sorry I misunderstood I see it now tagged on the gem. The gem data has a field called "stored_uses" which we can probably extract and use I will see if I can get it working.

Reply:
Here is a branch with the data. https://github.com/QuickStick123/PathOfBuilding/commit/683f2938e5a4417225d94648ab87739df8bc4be3

Reply:
I think it would be better to add the specialized code to the main code block instead of adding another switch to `getTriggerActionTriggerRate` . I'll have to rewrite good chunk of this to make it work with #4599 anyway so i doesn't matter much though.